mergify_engine/queue/merge_train.py

```diff
@@ -645,9 +649,33 @@ async def _set_initial_state(
             self.train.repository.log,
             False,
         )
+
+        if queue_rule.config["checks_timeout"] is not None:
+            await self.set_checks_timeout_context(queue_rule.config["checks_timeout"])
+
         await self.update_state(check_api.Conclusion.PENDING, evaluated_queue_rule)
         await self.update_summaries(check_api.Conclusion.PENDING, force_refresh=True)
 
+    async def set_checks_timeout_context(self, timeout: datetime.timedelta) -> None:
+        self.checks_timeout = timeout
+
+        if self.queue_pull_request_number is not None:
+            # Circular import
+            from mergify_engine import delayed_refresh
```

Comment:
😢 , I'm surprised we need that, but we can try to break the loop later.

Reply:
I tried to remove it but could'nt find a workaround, there another place also where this is used in `merge_train.py`